gensim/parsing/porter.py

```diff
@@ -98,39 +132,147 @@ def _m(self):
+        >>> p.j = 3
+        >>> p._doublec(3)
+        False
+
+        >>> from gensim.parsing.porter import PorterStemmer
+        >>> p = PorterStemmer()
+        >>> p.b = "really"
+        >>> p.j = 5
+        >>> p._doublec(4)
+        True
+
+        """
         return j > 0 and self.b[j] == self.b[j - 1] and self._cons(j)
 
     def _cvc(self, i):
-        """True <=> i-2,i-1,i has the form consonant - vowel - consonant
-        and also if the second c is not w,x or y. This is used when trying to
-        restore an e at the end of a short word, e.g.
+        """Check if b[i - 2], b[i - 1], b[i] have the form consonant letter - vowel letter- consonant letter
+        and also if the second 'c' is not 'w', 'x' or 'y'. This is used when trying to restore an 'e'
+        at the end of a short word, e.g. cav(e), lov(e), hop(e), crim(e),
+        but snow, box, tray.
+
+        Parameters
+        ----------
+        i : int
+            Index for `b`
+
+        Returns
+        -------
+        bool
+
+        Examples
+        --------
+        >>> from gensim.parsing.porter import PorterStemmer
+        >>> p = PorterStemmer()
+        >>> p.b = "lib"
+        >>> p.j = 2
+        >>> p._cvc(2)
+        True
+
+        >>> from gensim.parsing.porter import PorterStemmer
+        >>> p = PorterStemmer()
+        >>> p.b = "dll"
+        >>> p.j = 2
+        >>> p._cvc(2)
+        False
+
+        >>> from gensim.parsing.porter import PorterStemmer
+        >>> p = PorterStemmer()
+        >>> p.b = "wow"
+        >>> p.j = 2
+        >>> p._cvc(2)
+        False
 
-           cav(e), lov(e), hop(e), crim(e), but
-           snow, box, tray.
         """
         if i < 2 or not self._cons(i) or self._cons(i - 1) or not self._cons(i - 2):
             return False
         return self.b[i] not in "wxy"
 
     def _ends(self, s):
-        """True <=> 0,...k ends with the string s."""
+        """Check if sequence of letters b[0], ... , b[k] ends with the string `s`.
+
+        Parameters
+        ----------
+        s : str
+            Input string.
+
+        Returns
+        -------
+        bool
+
+        Examples
+        --------
+        >>> from gensim.parsing.porter import PorterStemmer
+        >>> p = PorterStemmer()
+        >>> p.b = "cowboy"
+        >>> p.j = 5
+        >>> p.k = 2
+        >>> p._ends("cow")
+        True
+
+        """
         if s[-1] != self.b[self.k]:  # tiny speed-up
-            return 0
+            return False
         length = len(s)
         if length > (self.k + 1):
-            return 0
+            return False
         if self.b[self.k - length + 1:self.k + 1] != s:
-            return 0
+            return False
         self.j = self.k - length
-        return 1
+        return True
 
     def _setto(self, s):
-        """Set (j+1),...k to the characters in the string s, adjusting k."""
+        """Set (j + 1), ... , k based on the characters from the string `s`, adjusting k.
```

Comment:
Double \`\`

Also, it simply appends \`\`s\`\` to the \`\`b\`\`. The description is kinda cryptic.